### DISCUSSION TITLE
Local File System Backup / Blob Container Backup

### DISCUSSION THREAD

**Discussion by PeterBrownIronMan** [requarks/wiki#3419] [Help / Questions]
**Question**
The Local File System and Blob Container Backup are setup and active but when I check the container folder there is no data.
1. For Local File System Backup does the blob container need to be set with security of public as no Account Name or Key is requested?
2. I've setup the Blob Container Backup with the appropriate details it requests and ran the manual back up, I get a status of Synchronised but the container is still empty?

**Host Info (please complete the following information):**
OS: Docker Container (Linux)
Wiki.js version: 2.4.107
Database engine: Postgres 10.11

**Comment by NGPixel** [comment]
Check the logs (`docker logs wiki`) for errors when running the manual dump.

**Comment by danielbarla** [comment]
I believe I may be running into the same issue, at least for Local File System.  It seems that Wiki.js is attempting to create the backup folder, but is failing:

> 2020-08-03T09:07:03.358Z [MASTER] warn: EACCES: permission denied, mkdir '/temp'

Somewhat oddly, it seems to be running `mkdir` on `/temp` even if the folder exists, and all users have write permissions to it (via `docker exec`, and a manual `mkdir` followed by a `chmod 777`).  I would be very interested in a solution or workaround!  For completeness, I'm also on version 2.4.107, using the standard requarks/wiki:2 image with hash e39c938d01db.

Minor update: as a shot in the dark, I've attempted to set up the backup folder in a location where presumably the process already has some access, specifically `/data/backups`.  I then manually gave all users access to this folder.  Unfortunately, the logs now show a similar error, but for `/data`, so this potential workaround failed.

**Comment by PeterBrownIronMan** [comment]
My Docker is on azure.
I'm not getting any logs created, the docker logs show nothing and refreshing doesn't generate any. Do you have any guides on how to setup Azure Blob Storage backup as I have this setup and this also doesn't create a backup in the blob storage. The Wiki.JS docs site doesn't have a page for help on this yet the option is available in the current version of Wiki.js ?

**Comment by mlabrum** [comment]
This is the error I'm getting

```
2020-08-25T02:02:16.642Z [MASTER] [32minfo[39m: (STORAGE/AZURE) Exporting all content to Azure Blob Storage...
2020-08-25T02:02:16.649Z [MASTER] [32minfo[39m: (STORAGE/AZURE) Adding page home.html...
2020-08-25T02:02:16.676Z [MASTER] [32minfo[39m: (STORAGE/AZURE) Adding asset wiki.svg...
2020-08-25T02:02:16.691Z [MASTER] [33mwarn[39m: ﻿<?xml version="1.0" encoding="utf-8"?><Error><Code>AuthenticationFailed</Code><Message>Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.
RequestId:5ff02289-e01e-008b-0883-7a691b000000
Time:2020-08-25T02:02:16.6860864Z</Message><AuthenticationErrorDetail>The MAC signature found in the HTTP request 'xkg7phg/FpV4jd7I3vIBzFsXuNls4PLYLpL8jWZjp8c=' is not the same as any computed signature. Server used following string to sign: 'PUT
application/octet-stream
x-ms-access-tier:Hot
x-ms-blob-type:BlockBlob
x-ms-client-request-id:034f8cce-8e1d-41d2-8959-84b18c4d5f0c
x-ms-date:Tue, 25 Aug 2020 02:02:16 GMT
x-ms-version:2019-07-07
/polyvisionwiki/wiki/wiki.svg'.</AuthenticationErrorDetail></Error>
```
